couchbase/src/main/java/org/talend/components/couchbase/service/CouchbaseService.java

```
@@ -0,0 +1,106 @@
+
+    @Service
+    private I18nMessage i18n;
+
+    public String[] resolveAddresses(String nodes) {
+        String[] addresses = nodes.replaceAll(" ", "").split(",");
+        for (int i = 0; i < addresses.length; i++) {
+            log.info(i18n.bootstrapNodes(i, addresses[i]));
+        }
+        return addresses;
+    }
+
+    public Bucket openConnection(CouchbaseDataStore dataStore) throws Exception {
+        String bootStrapNodes = dataStore.getBootstrapNodes();
+        String bucketName = dataStore.getBucket();
+        String password = dataStore.getPassword();
+
+        String[] urls = resolveAddresses(bootStrapNodes);
+
+        this.environment = new DefaultCouchbaseEnvironment.Builder().connectTimeout(20000L).build();
+        this.cluster = CouchbaseCluster.create(environment, urls);
+        this.bucket = this.cluster.openBucket(bucketName, password);
+        return bucket;
+    }
+
+    @HealthCheck("healthCheck")
+    public HealthCheckStatus healthCheck(@Option("configuration.dataset.connection") final CouchbaseDataStore datastore) {
+        try {
+            bucket = openConnection(datastore);
+            return new HealthCheckStatus(HealthCheckStatus.Status.OK, "Connection OK");
+        } catch (Exception exception) {
+            return new HealthCheckStatus(HealthCheckStatus.Status.KO, exception.getMessage());
+        } finally {
+            closeConnection();
+        }
+    }
+
+    public void closeConnection() {
+        if (bucket != null) {
+            if (bucket.close()) {
+                log.debug(i18n.bucketWasClosed(bucket.name()));
+            } else {
+                log.debug(i18n.cannotCloseBucket(bucket.name()));
+            }
+            bucket = null;
+        }
+        if (cluster != null) {
+            if (cluster.disconnect()) {
+                log.debug(i18n.clusterWasClosed());
+            } else {
+                log.debug(i18n.cannotCloseCluster());
```

no more error ? ne need to raise to user ?

---

disconnect() method can return true or false if cluster wasn't close.
[https://docs.couchbase.com/sdk-api/couchbase-java-client-2.0.0/com/couchbase/client/java/Cluster.html#disconnect()](Couchbase doc)

---

"true once done and everything succeeded, false otherwise." : if 'false' should we raise an error to the user ?